conda/plugins/virtual_packages/archspec.py

```diff
@@ -2,21 +2,21 @@
 # SPDX-License-Identifier: BSD-3-Clause
 """Detect archspec name."""
 
-import os
-
+from ...auxlib import NULL
 from .. import hookimpl
 from ..types import CondaVirtualPackage
 
 
+def archspec_build_num():
+    from ...core.index import get_archspec_name
+
+    build_num = get_archspec_name()
+    return build_num or NULL
+
+
 @hookimpl
 def conda_virtual_packages():
-    archspec_name = os.getenv("CONDA_OVERRIDE_ARCHSPEC")
-    if archspec_name is None:  # no override found
-        from ...core.index import get_archspec_name
-
-        archspec_name = get_archspec_name()
-        if archspec_name:
-            yield CondaVirtualPackage("archspec", "1", archspec_name)
-    elif archspec_name:  # truthy override found
-        yield CondaVirtualPackage("archspec", "1", archspec_name)
+    yield CondaVirtualPackage(
+        name="archspec", version="1", build=archspec_build_num, override_entity="build"
```

Comment:
Adding trailing comma to force multiline:

```suggestion
        name="archspec",
        version="1",
        build=archspec_build_num,
        override_entity="build",
```